src/wasm/wasm-validator.cpp

```diff
@@ -795,10 +795,11 @@ void FunctionValidator::visitCall(Call* curr) {
         curr,
         "calls may only be unreachable if they have unreachable operands");
     } else {
-      shouldBeEqual(curr->type,
-                    target->sig.results,
-                    curr,
-                    "call type must match callee return type");
+      shouldBeSubTypeOrFirstIsUnreachable(
```

Comment:
What is the benefit of allowing `Call` expressions to have imprecise types?

Reply:
Oh, good point, it doesn't need to be a subtype...